.github/workflows/coder.yaml

```diff
@@ -288,30 +305,36 @@ jobs:
           - windows-2022
     steps:
       - uses: actions/checkout@v3
+        if: needs.changes.outputs.go == 'true'
```

Comment:
Sadly, because this job is required, we can't use this `if` at the top or the PR status will never show Success. Alternative approach linked in PR discussion which involves creating a generic workflow with same name and running it for the opposition condition (i.e. no Go changes). I like this approach better but could be swayed the other way.